### DISCUSSION TITLE
voice file size and info?

### DISCUSSION THREAD

**Discussion by radiohound** [pschatzmann/arduino-espeak-ng#4] [General]
Hello, I have a few questions about the great work you have done on this. 

How did you reduce the file size of the binary file that is normally called en2, or one of the other voice files that are all normally 7.2 megabytes, down to what I think is phonedata binary file that is about 500 kilobytes? 

Is there any way to pull the sound files for each phoneme from either of these binary files to individual sound files? If so, how could I go about doing it?

Thank you!
Walter

**Comment by pschatzmann** [comment]
I don't understand where you have the 7.2mb from. As you can see in the Readme the program and data only uses 1.6 MB PROGMEM for English. An ESP32 usually has 4MB (with versions with 8 or 16MB) so you have enough headroom for your own sketch...

I am not really knowledgeable, but as far as I know you can create your own phonedata files: https://github.com/espeak-ng/espeak-ng/blob/master/docs/add_language.md

**Reply by radiohound** [reply]
Thank you! Not sure where I got 7.2 mb from either. I was able to find the bare phoneme wave files in espeak-ng-master/phsource/ directories, like espeak-ng-master/phsource/ustop . Which is what I was trying to do. Thanks again.

**Comment by pschatzmann** [comment]
Standard espeak-ng is loading the phoneme files into RAM. This is exactly what I avoided to do because there is no enough RAM on any microcontroller to do this! 

Instead I am loading them into PROGMEM from where I can just access them with the file API w/o using any RAM at all...
